routing-daemon/lib/openshift/routing/controllers/batched.rb

```diff
@@ -104,10 +104,10 @@ def create_monitor monitor_name, path, up_code, type, interval, timeout
       @monitors.push monitor_name
     end
 
-    def delete_monitor monitor_name, pool_name=nil
+    def delete_monitor monitor_name, pool_name=nil, type
       raise LBControllerException.new "Monitor not found: #{monitor_name}" unless monitors.include? monitor_name
 
-      @lb_model.delete_monitor monitor_name if monitors.include? monitor_name
+      @lb_model.delete_monitor monitor_name, type if monitors.include? monitor_name
```

Comment:
This line shows that things are kind of tricky now.  Is it safe to assume that there will never be two monitors with the same name but different types? If not, we'll need to change `monitors` from an array of monitor names to an array of tuples of monitor name and monitor type and then modify this check accordingly.